routes/bots.js

```diff
@@ -5,104 +5,171 @@ const bot = require("../utils/discordbot");
+      const b = req.body;
+      if (!(await captchaIsValid(config.recaptcha, b["g-recaptcha-response"])))
+        return res.render("message", {
+          message: "O Captcha precisa ser validado.",
+          url: req.originalUrl,
+        });
+      const allTags = Object.keys(tags);
+      const botTags = [
+        ...new Set(typeof b.tags == "string" ? [b.tags] : b.tags),
+      ];
+      const owners = b.owners ? [...b.owners] : null;
+      {
+        if (!owners || owners.some((o) => isNaN(o) || o.length != 18))
+          return res.render("message", {
+            message: "Lista de donos inválida.",
+            url: req.originalUrl,
+          });
+
+        if (isNaN(b.id) || b.id.length != 18)
+          return res.render("message", {
+            message: "ID do bot fornecido é inválido.",
+            url: req.originalUrl,
+          });
+        if (!b.library || !libraries.includes(b.library))
+          return res.render("message", {
+            message: "Biblioteca fornecida é inválida.",
+            url: req.originalUrl,
+          });
+        if (b.support && b.support.length > 2083)
+          return res.render("message", {
+            message: "Link do servidor de suporte é inválido.",
+            url: req.originalUrl,
+          });
+        if (!b.prefix || b.prefix.length > 15)
+          return res.render("message", {
+            message: "Prefixo do bot é inválido.",
+            url: req.originalUrl,
+          });
+        if (!b.shortdesc || b.shortdesc.length < 2 || b.shortdesc.length > 300)
+          return res.render("message", {
+            message: "Descrição curta é inválida.",
+            url: req.originalUrl,
+          });
+        if (b.longdesc && b.longdesc.length > 10000)
+          return res.render("message", {
+            message: "Descrição longa é inválida.",
+            url: req.originalUrl,
+          });
+        if (
+          !botTags.length ||
+          botTags.length > 6 ||
+          botTags.some((t) => !allTags.includes(t))
         )
-            return res.render("message", {message: "Algo de errado não está certo.", url: req.originalUrl});
-        
-        dBot.fetchUser(b.id).then(user => {
-            if (!user)
-                return res.render("message", {message: "O ID fornecido é inválido.", url: req.originalUrl});
-            
-            if (!user.bot)
-                return res.render("message", {
-                    message: `O ID fornecido pertence a ${userToString(user)}, que não é um bot.`,
-                    url: req.originalUrl
-                });
-            
-            db.Bots.findById(b.id).exec().then(async dbUser => {
-                if (dbUser)
-                    return res.render("message", {
-                        message: `O ID fornecido pertence a ${userToString(user)}, que já está cadastrado no sistema.`,
-                        url: req.originalUrl
-                    });
-                if (owners.size) {
```

Comment:
Not needed....

Owners is always a list

Reply:
This was to make it compatible with the current BPD database, which is not always a list

Reply:
But you've converted as list
Def: https://github.com/zuraaa-projects/Zuraaa.com/blob/2925fff76971d9c125625a2977a2507b46c8dce4/routes/bots.js#L30

Using: https://github.com/zuraaa-projects/Zuraaa.com/blob/2925fff76971d9c125625a2977a2507b46c8dce4/routes/bots.js#L32

😕